### PR TITLE
scaffold: ai

### LINKED ISSUE #26: integrate: openai
TODO:

- [ ]  Write unit tests
- [x] Write openai service methods
- [ ]  Write health check methods

### PR DESCRIPTION
**Pull request by Nasfame**
## Related issues

Fix #26 #27

## Proposed Changes

[//]: # ([List down the specific changes you have made in this pull request.])

- [x]  Add scaffold for openai
- [x] Add scaffold for pinecone

## Checklist

- [x] I have followed the repository's contribution guidelines.
- [x] I have added necessary documentation or updated existing documentation.
- [x] My code follows the established code style and formatting of the project.
- [x] I have added suitable unit tests (if applicable).

## To-Do

- [ ]  Write unit tests
- [ ] Write openai service methods
- [ ] Write pinecone service methods

## API Changes documentation

```shell


```

```shell


```

### CODE REVIEW

File: src/openai/openai.service.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { Injectable } from '@nestjs/common';
+import { Configuration, CreateCompletionRequest, OpenAIApi } from "openai";
+const MODULE_ID="text-davinci-003"
+const defaultTemperature=0.9
+@Injectable()
+export class OpenaiService {
+    private readonly OpenaiApi:OpenAIApi
+    constructor(){
+        const configuration = new Configuration({
+            organization: process.env.example.ORGANISATION_ID,
```
Comment (Nasfame) [inline src/openai/openai.service.ts:10]: Use configService. Refer prismaService

File: src/openai/openai.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [OpenaiService],
+    }).compile()
+
+    service = module.get<OpenaiService>(OpenaiService)
+  })
+
+  it('should be defined', () => {
+    expect(service).toBeDefined() //TODO:
```
Comment (Nasfame) [inline src/openai/openai.service.spec.ts:16]: Define tests for each service method

**Review by Nasfame — COMMENTED** [review]
Fix the bugs

File: src/openai/openai.service.ts (hunk 1 above)
Comment (mhasif0786) [reply src/openai/openai.service.ts]: This is done

File: src/openai/openai.service.ts [hunk 3]
```diff
@@ -1,15 +1,17 @@
...
 export class OpenaiService {
     private readonly OpenaiApi:OpenAIApi
-    constructor(){
-        const configuration = new Configuration({
-            organization: process.env.example.ORGANISATION_ID,
-            apiKey: process.env.example.OPENAI_API_KEY,
-        });
+    constructor(private readonly config:ConfigModule){
+        const configuration = [{
+            organization: config.get<string>('ORGANISATION_ID'),
```
Comment (Nasfame) [inline src/openai/openai.service.ts:11]: Use OPENAI_ORGANISATION_ID

**Comment by Nasfame** [comment]
Done in #47

**Pull request closed without merging**